src/js/wrapper.ts

```diff
@@ -150,13 +152,19 @@ export const NATIVE: SentryNativeWrapper = {
       (itemHeader as BaseEnvelopeItemHeaders).length = bytesPayload.length;
       const serializedItemHeader = JSON.stringify(itemHeader);
 
-      envelopeBytes.push(...utf8ToBytes(serializedItemHeader));
-      envelopeBytes.push(EOL);
-      envelopeBytes = envelopeBytes.concat(bytesPayload);
-      envelopeBytes.push(EOL);
+      const bytesItemHeader = utf8ToBytes(serializedItemHeader);
+      const newBytes = new Uint8Array(
+        envelopeBytes.length + bytesItemHeader.length + EOL.length + bytesPayload.length + EOL.length,
+      );
+      newBytes.set(envelopeBytes);
+      newBytes.set(bytesItemHeader, envelopeBytes.length);
+      newBytes.set(EOL, envelopeBytes.length + bytesItemHeader.length);
+      newBytes.set(bytesPayload, envelopeBytes.length + bytesItemHeader.length + EOL.length);
+      newBytes.set(EOL, envelopeBytes.length + bytesItemHeader.length + EOL.length + bytesPayload.length);
+      envelopeBytes = newBytes;
     }
 
-    await RNSentry.captureEnvelope(envelopeBytes, { store: hardCrashed });
+    await RNSentry.captureEnvelope(fromByteArray(envelopeBytes), { store: hardCrashed });
```

Comment:
M: Instead of passing a base64 string, why don't we pass the stringified envelope/JSON? that's how it used to be before moving to byte array.

Reply:
The main reason to move to the byte array was attachments, as those come as a byte array (or string).

Second, by assembling the byte array in JS we avoid possible mistakes in the byte length. Originally the string we send to the native layer for calculation of the byte length.

We could change to some hybrid solution of creating JSON envelopes without the lengths and only encoding attachments. The `Uint8Array` either must be copied to `int array` or encoded into a string. For large attachments 5-20 MB this takes roughly the same time.

Reply:
If they actually take roughly the same time, we'd spare vendoring the base64 class at least (bundle size).
Your call here, just trying to ponder the trade-offs.

Reply:
True, I've not thought about the bundle size.